### ISSUE TITLE
Styling makes text garbled or missing when using OTF or TTF fonts

### ISSUE THREAD

**Issue by RGBz** [rkusa/pdfjs#81]
There seems to be some issue when changing style from one TTF/OTF font to another within the same text block using `add()`. The code below generates a PDF that shows the first part of the text missing and then only "**eating a quarter pack per day.**" Styling seems to work fine when using the AFM fonts provided, but renders missing or garbled text when switching between other kinds of fonts.

```
const fs = require('fs');
const path = require('path');
const pdf = require('pdfjs');
const HARMONIA = new pdf.Font(fs.readFileSync(path.join(__dirname, 'fonts', 'HarmoniaSansProCyr-Light.otf')));
const HARMONIA_B = new pdf.Font(fs.readFileSync(path.join(__dirname, 'fonts', 'HarmoniaSansProCyr-Black.otf')));

const doc = new pdf.Document({ font: HARMONIA, padding: 10 })
doc.pipe(fs.createWriteStream('output.pdf'))

doc.text()
  .add('Manage your calories by', { font: HARMONIA })
  .add('eating a quarter pack per day.', { font: HARMONIA_B });

doc.end();
```

**Comment by rkusa** [comment]
Oops, shame on me for not having this covert with tests before. Thanks for the report! Should be fixed in `2.0.0-alpha.4`

**Comment by RGBz** [comment]
Fantastic that fixed everything, thanks so much! Great project!
